metadata-service/auth-impl/src/main/java/com/datahub/authorization/DataHubAuthorizer.java

```diff
@@ -184,15 +189,32 @@ private boolean isRequestGranted(final DataHubPolicyInfo policy, final Authoriza
     if (AuthorizationMode.ALLOW_ALL.equals(mode())) {
       return true;
     }
+
+    Optional<Urn> actorUrn = getUrnFromRequestActor(request.getActorUrn());
+    if (actorUrn.isEmpty()) {
```

Comment:
Why would this happen? Have we seen this case occur, where actorUrn is null in the request?

Reply:
This logic was moved from the [PolicyEngine class](https://github.com/datahub-project/datahub/blob/master/metadata-service/auth-impl/src/main/java/com/datahub/authorization/PolicyEngine.java#L50-L63) to here (due to the decision of passing the resolved spec which need a urn).